service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -651,13 +649,14 @@ private void collectEventAboutConfirmWaste(Map<Integer, Integer> confirmed, Orde
         Long orderId, int countOfChanges, StringBuilder values) {
         for (Map.Entry<Integer, Integer> entry : confirmed.entrySet()) {
             Integer capacity = bagRepository.findCapacityById(entry.getKey());
-            BagTranslation bagTranslation = bagTranslationRepository.findBagTranslationByBagId(entry.getKey());
+            Optional<Bag> bagOptional = bagRepository.findById(entry.getKey());
 
-            if (order.getOrderStatus() == OrderStatus.ADJUSTMENT
+            if (bagOptional.isPresent() && order.getOrderStatus() == OrderStatus.ADJUSTMENT
                 || order.getOrderStatus() == OrderStatus.CONFIRMED
                 || order.getOrderStatus() == OrderStatus.FORMED
                 || order.getOrderStatus() == OrderStatus.NOT_TAKEN_OUT) {
                 Optional<Long> confirmWasteWas = Optional.empty();
+                Bag bag = bagOptional.isPresent() ? bagOptional.get() : Bag.builder().build();
```

Comment:
It is redundant `bagOptional.isPresent()`
You already have chacked it in the `if`

Reply:
I made that because sonarcloud detected it as bad code or bug, I don't remember what exactly it was

Reply:
I had something like this: Bag bag = bagOptional.get()
But sonar cloud detect that bagOption may be null and that it's an error.
What sould I do?